File: calls_app/android/app/src/main/kotlin/com/example/in_app_calls_demo/models/CallData.kt
package com.example.in_app_calls_demo.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
class CallData(
        val channelId: String?,
        val callerId: String?,
        val callerPhone: String?,
        val callerName: String?,
        val hasVideo: Boolean?,
) : Parcelable {
    fun hasVideoToInt(): Int {
        if (hasVideo == true) return 1
        return 0
    }

    fun toMap(): HashMap<String, Any?> {
        return hashMapOf(
                "channelId" to channelId,
                "callerId" to callerId,
                "callerPhone" to callerPhone,
                "callerName" to callerName,
                "hasVideo" to hasVideo
        )
    }
}

File: calls_app/lib/data/models/call_data.dart
class CallData {
  CallData({
    required this.channelId,
    required this.callerId,
    required this.callerPhone,
    required this.callerName,
    required this.hasVideo,
  });

  factory CallData.fromMap(Map<String, dynamic> data) {
    return CallData(
      channelId: data['channelId'] as String?,
      callerId: data['callerId'] as String?,
      callerPhone: data['callerPhone'] as String?,
      callerName: data['callerName'] as String?,
      hasVideo: data['hasVideo'] as bool?,
    );
  }

  final String? channelId;
  final String? callerId;
  final String? callerPhone;
  final String? callerName;
  final bool? hasVideo;
}
